doc/glm/first_level_model.rst

```diff
@@ -0,0 +1,178 @@
+Nilearn offers a few different HRF models including the commonly used double-gamma SPM model ('spm') and the model shape proposed by G. Glover ('glover'), both allowing the option of adding time and dispersion derivatives. The addition of these derivatives allows to better model any uncertainty in timing information. In addition, an FIR (finite impulse response, 'fir') model of the HRF is also available.
+
+In order to visualize the predicted regressor prior to plugging it into the linear model, use the function :func:`nilearn.stats.first_level_model.compute_regressor`, or explore the HRF plotting example: :ref:`sphx_glr_auto_examples_04_glm_first_level_models_plot_hrf.py`
+
+
+Design matrix: event-based and time series-based
+================================================
+
+Event-based
+-----------
+
+To create an event-based design matrix, information about the trial type, onset time and duration of the events in the experiment are necessary. This can be provided by the user, or be part of the dataset if using a BIDS-compatible dataset or one of the nilearn dataset fetcher functions like :func:`nilearn.datasets.fetch_spm_multimodal_fmri`, :func:`nilearn.datasets.fetch_language_localizer_demo_dataset`, etc. Using a nilearn fetcher function, e.g. :func:`nilearn.datasets.fetch_spm_multimodal_fmri`, first download the data (for one subject)::
+
+  from nilearn.datasets import fetch_spm_multimodal_fmri
+  subject_data = fetch_spm_multimodal_fmri()
+
+Here, each subject has 2 sessions of data; events for the first session can be accessed using::
+
+  import pandas as pd
+  events = pd.read_table(subject_data['events{}'.format(1)])
+
+If the events details are provided by the user, then details about the experiment should be provided. This information can be stored in a DataFrame and used instead of the events file::
+
+  import pandas as pd
+  conditions = ['c0', 'c0', 'c0', 'c1', 'c1', 'c1', 'c3', 'c3', 'c3']
+  duration = [1., 1., 1., 1., 1., 1., 1., 1., 1.]
+  onsets = [30., 70., 100., 10., 30., 90., 30., 40., 60.]
+  events = pd.DataFrame({'trial_type': conditions, 'onset': onsets, 'duration': duration})
+
+In addition, some scan attributes also need to be specified::
+
+    import numpy as np
+    tr = 1.0  # repetition time is 1 second
+    n_scans = 128  # the acquisition comprises 128 scans
+    frame_times = np.arange(n_scans) * tr  # here are the corresponding frame times
+
+In either case, the design matrix is created using the :func:`nilearn.stats.first_level_model.make_first_level_design_matrix` function::
+
+  from nilearn.stats.first_level_model import make_first_level_design_matrix
+  design_matrices = make_first_level_design_matrix(frame_times, events,
+                            drift_model='polynomial', drift_order=3)
+
+A handy function called :func:`nilearn.reporting.plot_design_matrix()` can be used to visualize the design matrix::
+
+  from nilearn.reporting import plot_design_matrix
+  plot_design_matrix(design_matrices)
```

Comment:
All this should be in an example tutorial. Only the big-picture ideas should be here.

Reply:
I disagree with this. This should be common practice so have to be mentionned

Reply:
I agree with @thomasbazeille on this. I've modified the text to make it sound less code-y.

Reply:
Thx !